### PR TITLE
chore!: upgrading to Java 17

### PR DESCRIPTION
**Pull request by geovanne-duarte-mx**
Updating build for Java 17

### CODE REVIEW

**Review by gerfboy — APPROVED** [review]

**Review by gerfboy — CHANGES_REQUESTED** [review]
When will we bump the version?

File: build.gradle [hunk 1]
```diff
@@ -4,30 +4,28 @@ plugins {
   id "java-gradle-plugin"
   id "java-library"
   id "maven-publish"
-  id "com.github.ben-manes.versions" version "0.43.0"
+  id "com.github.ben-manes.versions" version "0.51.0"
 }
 
 group "com.mx.coppuccino"
 version "3.3.0" // x-release-please-version
```
Comment (gerfboy) [inline build.gradle:11]: Are we thinking of bumping this to 4.0?

File: build.gradle [hunk 2]
```diff
@@ -4,30 +4,28 @@ plugins {
...
-sourceCompatibility = 1.8
-targetCompatibility = 1.8
+sourceCompatibility = JavaVersion.VERSION_17
```
Comment (gerfboy) [inline build.gradle:12]: You'll definitely need to talk to Matt N about if we're ready for this changed.

File: build.gradle (hunk 1 above)
Comment (mattnichols) [reply build.gradle:11]: semantic-release will do that for us.

File: build.gradle (hunk 2 above)
Comment (mattnichols) [reply build.gradle:12]: I wouldn't do this yet.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -16,4 +16,3 @@ jobs:
     uses: mxenabled/path-tools/.github/workflows/ci.yml@master
     with:
       force: ${{ inputs.force != '' && inputs.force }}
-      java-version: '8'
```
Comment (mattnichols) [inline .github/workflows/ci.yml:19]: ??

**Review by meotch — APPROVED** [review]

**Review by gerfboy — APPROVED** [review]

**Pull request merged**